app/views/SearchMessagesView/index.tsx

```diff
@@ -153,12 +152,15 @@ class SearchMessagesView extends React.Component<ISearchMessagesViewProps, ISear
 				.fetch();
 		}
 		// If it's not a encrypted room, search messages on the server
-		const result = await RocketChat.searchMessages(this.rid, searchText, QUERY_SIZE, this.offset);
-		if (result.success) {
-			return result.messages;
-		}
-
-		return [];
+		const whereClause = [Q.where('rid', this.rid), Q.where('msg', Q.like(`%${likeString}%`)), Q.experimentalTake(QUERY_SIZE)] as (
+			| Q.WhereDescription
+			| Q.Or
+		)[];
+		const messagesCollection = db
+			.get('messages')
+			.query(...whereClause)
+			.fetch();
+		return messagesCollection;
```

Comment:
The problem with this approach is that you're using the local DB when searching for each term so you're just going to receive messages you have saved on your device; next time that you probably clean server cache you might not receive as many messages, or any message at all. 

I would suggest to check how our markdown system works (the one inside the `markdown/new` folder). Some resource that might help a bit is [how the web currently does it](https://github.com/RocketChat/Rocket.Chat/tree/f0c37c5a8c904181577b80d239255284ac6a1280/apps/meteor/client/components/Message/MessageBodyRender), but in our case we might need extra validations or styling because mobile works differently in some cases, and I believe this might be one of them.

Reply:
Ohh okay, I got the gist of it 
The issue with the method that was being used earlier is that there is a difference in the message content format that RocketChat search api returns and what we get when we search local database.
When we render message in the RoomView we take the messages from database only, so the MessageContainer is designed to handle input of data format as used by the database.
So here I tried to fetch messages as it is done in the RoomView.

Reply:
@gerzonc I have changed the implementation, can you review the latest pr, now rather than changing the data retrieval source, I am changing the format after retrieving it from the server.
<img width="258" alt="Screenshot 2022-04-15 at 11 17 12 PM" src="https://user-images.githubusercontent.com/55805094/163603768-1cd13f47-98fa-4a67-8429-74f55b1dce73.png">